File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/group/chat/GroupHistoryPinMessageResponse.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.group.chat;

import com.zfoo.app.zapp.common.protocol.common.ChatMessage;
import com.zfoo.protocol.IPacket;

import java.util.List;

/**
 * 好友聊天
 *
 * @author jaysunxiao
 * @version 1.0
 * @since 2019-11-14 10:54
 */
public class GroupHistoryPinMessageResponse implements IPacket {

    public static final transient short PROTOCOL_ID = 18113;

    private long groupId;
    private long channelId;

    private long lastMessageId;

    private List<ChatMessage> chatMessages;

    public static GroupHistoryPinMessageResponse valueOf(long groupId, long channelId, long lastMessageId, List<ChatMessage> chatMessages) {
        var response = new GroupHistoryPinMessageResponse();
        response.groupId = groupId;
        response.channelId = channelId;
        response.lastMessageId = lastMessageId;
        response.chatMessages = chatMessages;
        return response;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }

    public long getGroupId() {
        return groupId;
    }

    public void setGroupId(long groupId) {
        this.groupId = groupId;
    }

    public long getChannelId() {
        return channelId;
    }

    public void setChannelId(long channelId) {
        this.channelId = channelId;
    }

    public long getLastMessageId() {
        return lastMessageId;
    }

    public void setLastMessageId(long lastMessageId) {
        this.lastMessageId = lastMessageId;
    }

    public List<ChatMessage> getChatMessages() {
        return chatMessages;
    }

    public void setChatMessages(List<ChatMessage> chatMessages) {
        this.chatMessages = chatMessages;
    }
}

File: web/src/jsProtocol/group/chat/GroupHistoryPinMessageResponse.js
// 好友聊天
//
// @author jaysunxiao
// @version 1.0
// @since 2019-11-14 10:54
const GroupHistoryPinMessageResponse = function(channelId, chatMessages, groupId, lastMessageId) {
    this.channelId = channelId; // long
    this.chatMessages = chatMessages; // java.util.List<com.zfoo.app.zapp.common.protocol.common.ChatMessage>
    this.groupId = groupId; // long
    this.lastMessageId = lastMessageId; // long
};

GroupHistoryPinMessageResponse.prototype.protocolId = function() {
    return 18113;
};

GroupHistoryPinMessageResponse.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
    buffer.writeLong(packet.channelId);
    buffer.writePacketArray(packet.chatMessages, 120);
    buffer.writeLong(packet.groupId);
    buffer.writeLong(packet.lastMessageId);
};

GroupHistoryPinMessageResponse.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new GroupHistoryPinMessageResponse();
    const result0 = buffer.readLong();
    packet.channelId = result0;
    const list1 = buffer.readPacketArray(120);
    packet.chatMessages = list1;
    const result2 = buffer.readLong();
    packet.groupId = result2;
    const result3 = buffer.readLong();
    packet.lastMessageId = result3;
    return packet;
};

export default GroupHistoryPinMessageResponse;
